ansible_collections/arista/avd/docs/porting-guides/5.x.x.md

```diff
@@ -315,9 +321,88 @@ The description can be reverted manually if needed:
+
+In AVD 5.0.0 setting the `description` key under `port_channel` of the adapter will replace the full port-channel interface description.
+The description fields support templates. See the input variables documentation for available fields.
+
+To retain the old descriptions, the value of the `description` key under `port_channel` must be *moved* to the new key
+`endpoint_port_channel` for *all* adapters. Also the default description template must be updated.
+
+```diff
+ servers:
+   - name: server01
+     adapters:
+       - <...>
+         description: My adapter description
+         port_channel:
+-          description: My description that was added after the adapter description
++          endpoint_port_channel: My description that was added after the adapter description
+
++default_connected_endpoints_port_channel_description: "{adapter_description_or_endpoint}{endpoint_port_channel?<_}"
+```
+
+#### Network Ports ethernet interfaces
+
+The default interface description for the `network_ports` data model changes to `<endpoint>`, allowing
+to use new `endpoint` key under `network_ports` which is used similar to the endpoint name under connected endpoints.
+Previously there was no default description.
+
+Setting the description directly under `network_ports` still takes precedence.
+The description fields support templates. See the input variables documentation for available fields.
+
+The default descriptions can be reverted manually if needed:
+
+```diff
++default_network_ports_description: ""
+```
+
+#### Network Ports port-channel interfaces
+
+The default interface description for port-channels under the `network_ports` data model changes to `<endpoint>_<endpoint_port_channel>`,
+allowing to use new `endpoint` key under `network_ports` which is used similarly to the endpoint name under connected endpoints.
+Previously the description was `<network_port_description>_<port_channel_description>`, so the description field
+set under `port_channel` was always added after the `description` set directly under the `network_ports`.
+
+In AVD 5.0.0 setting the `description` key under `port_channel` of the network port will replace the full port-channel interface description.
+The description fields support templates. See the input variables documentation for available fields.
+
+To retain the old descriptions, the value of the `description` key under `port_channel` must be *moved* to the new key
+`endpoint_port_channel` for *all* network_ports. Also the default description template must be updated.
+
+```diff
+ network_ports:
+   - <...>
+     description: My ports description
+     port_channel:
+-      description: My description that was added after the adapter description
++      endpoint_port_channel: My description that was added after the adapter description
+
++default_network_ports_port_channel_description: "{adapter_description_or_endpoint}{endpoint_port_channel?<_}"
```

Comment:
Is this a valid syntax: `{adapter_description_or_endpoint}` ?

Reply:
yes there is a special helper field called that. It is needed to emulate the historic behavior, so making it easier to revert to 4.x style.